test/SlowTests/Issues/RavenDB-14541_includeTimeSeries.cs

```diff
@@ -0,0 +1,499 @@
+                }
+
+                using (var session = store.OpenSession())
+                {
+                    var query = from a in session.Query<User>()
+                        let _ = RavenQuery.IncludeAllCounters(a)
+                        let __ = RavenQuery.IncludeTimeSeries(a, "CurViews")
+                        let ___ = RavenQuery.Include<State>("a.StateId")
+                                select new { Name = a.UserName};
+
+                    var AsString = query.ToString();
+                    var result = query.ToList();
+
+                    var tuckerCounterGetter = session.CountersFor("user/tucker");
+                    var tuckerAllCounters = tuckerCounterGetter.GetAll();
+                    var tuckerShoes = tuckerCounterGetter.Get("shoes");
+
+                    var tuckerTimeSeries = session.TimeSeriesFor("user/tucker", "CurViews");
+
+                    var kevinState = session.Load<State>("states/2");
+
+                    Assert.Equal(1, session.Advanced.NumberOfRequests);
+
+                }
+            }
+        }
+
+        [Fact]
+        public void SessionQueryAllIncludevariantsInRavenQuery_ShouldThrow()
+        {
+            using (DocumentStore store = GetDocumentStore())
+            {
+                var baseline = DateTime.Today;
+
+                var start = baseline;
+                var end = baseline.AddHours(5);
+
+                using (var session = store.OpenSession())
+                {
+                    session.Store(new State { Name = "Alabama" }, "states/1");
+                    session.Store(new State { Name = "Minassota" }, "states/2");
+
+                    session.Store(new User { UserName = "Tucker", StateId = "states/1" }, "user/tucker");
+                    session.Store(new User { UserName = "Kevin", StateId = "states/2" }, "user/kevin");
+
+
+                    session.TimeSeriesFor("user/Tucker", "CurViews").Append(baseline.AddHours(2), 55, "views/Tucker");
+                    session.TimeSeriesFor("user/Tucker", "CurViews").Append(baseline.AddHours(3), 12, "views/Tucker");
+                    session.TimeSeriesFor("user/Tucker", "CurViews").Append(baseline.AddHours(4), 15, "views/Tucker");
+
+                    session.CountersFor("user/tucker").Increment("shoes", 100);
+                    session.CountersFor("user/tucker").Increment("shirts", 20);
+                    session.CountersFor("user/tucker").Increment("pants", 3);
+                    session.CountersFor("user/kevin").Increment("shoes", 5);
+
+                    session.SaveChanges();
+                }
+
+                using (var session = store.OpenSession())
+                {
+                    var query = from a in session.Query<User>()
+                                let _ = RavenQuery.IncludeAllCounters(a)
+                                let __ = RavenQuery.IncludeTimeSeries(a, "CurViews")
+                                let a_ = RavenQuery.Include<State>("a.StateId")
```

Comment:
why `let a_` is allowed? we should allow only discard variables

Reply:
I had a wrong regex, fixed the regex, and the test